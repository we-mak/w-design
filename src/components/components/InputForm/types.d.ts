import { ActionProps } from "../../../typings";
import { InputFieldProps } from "../../elements/InputField/types";

export interface InputFormProps extends InputFieldProps, ActionProps {
  label?: string;
  isError?: boolean;
  isWarning?: boolean;
  isSuccess?: boolean;
  hintMessage?: string;
}
