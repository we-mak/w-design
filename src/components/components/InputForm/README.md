| Props       | type    | default | description                                                     |
| ----------- | ------- | ------- | --------------------------------------------------------------- |
| id          | string  |         | Text field id attribute, automatically apply for label identify |
| label       | string  |         | Input label                                                     |
| hintMessage | string  |         | Text hint message for the input                                 |
| isRequired  | boolean | false   |                                                                 |
| isError     | boolean | false   |                                                                 |
| isWarning   | boolean | false   |                                                                 |
| isSuccess   | boolean | false   |                                                                 |
| children    | any     |         | other element appears below the input field                     |
| fluid       | boolean | false   |                                                                 |
