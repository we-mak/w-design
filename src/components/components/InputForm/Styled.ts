import styled from "styled-components";
import { InputFormProps } from "./types";

export const FieldGroup = styled.div<InputFormProps>`
  position: relative;
  &:not(:last-child) {
    margin-bottom: ${props => props.theme.space[2]}px;
  }

  i {
    color: ${props => props.theme.colors["N10"]};

    &.icon-before {
      height: 0.8rem;
      margin: 0 0.25rem;
      position: absolute;
      top: 73%;
      transform: translateY(-50%);
      width: 0.8rem;
      z-index: 2;
      left: 0.1rem;
    }
    &.icon-after {
      height: 0.8rem;
      margin: 0 0.25rem;
      position: absolute;
      top: 73%;
      transform: translateY(-50%);
      width: 0.8rem;
      z-index: 2;
      right: 0.1rem;
    }
  }
  input {
    &.icon-before {
      padding-left: 1.3rem !important;
    }
    &.icon-after {
      padding-right: 1.3rem !important;
    }
  }
`;
FieldGroup.displayName = "FieldGroup";

export const Label = styled.label`
  display: block;
  line-height: ${props => props.theme.lineHeights[0]};
  padding: ${props => props.theme.space[2]}px 0;
  font-weight: ${props => props.theme.fontWeights.strong};
`;
Label.displayName = "Label";

export const TextHint = styled.small<InputFormProps>`
  display: block;
  margin-top: 0.4rem;
  color: ${props => {
    const { isWarning, isError, isSuccess, theme } = props;
    if (isWarning) return `${theme.colors["O30"]}`;
    if (isError) return `${theme.colors["R30"]}`;
    if (isSuccess) return `${theme.colors["G30"]}`;
    return `${theme.colors["N30"]}`;
  }};
`;
TextHint.displayName = "TextHint";

export const RequiredMark = styled.sup`
  color: ${props => props.theme.colors["R30"]};
  font-size: ${props => props.theme.fontSizes[0]};
  top: -0.2rem;
`;

export const SpinnerWrap = styled.span`
  position: absolute;
  right: 1rem;
  top: 60%;
`;
