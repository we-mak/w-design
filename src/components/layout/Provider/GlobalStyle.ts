import { createGlobalStyle } from "styled-components";
import theme from "../../../common/styleUtils/theme";
const colors = theme.colors;

/*! normalize.css v8.0.1 | MIT License | github.com/necolas/normalize.css */
export const GlobalStyle = createGlobalStyle`
html {
  line-height: 1.15;
  -webkit-text-size-adjust: 100%;
}
body {
  margin: 0;
}
main {
  display: block;
}
h1 {
  font-size: 2em;
  margin: 0.67em 0;
}
hr {
  box-sizing: content-box;
  height: 0;
  overflow: visible;
}
pre {
  font-family: monospace, monospace;
  font-size: 1em;
}
a {
  background-color: transparent;
}
abbr[title] {
  border-bottom: none;
  text-decoration: underline;
  text-decoration: underline dotted;
}
b,
strong {
  font-weight: bolder;
}
code,
kbd,
samp {
  font-family: monospace, monospace;
  font-size: 1em;
}
small {
  font-size: 80%;
}
sub,
sup {
  font-size: 75%;
  line-height: 0;
  position: relative;
  vertical-align: baseline;
}
sub {
  bottom: -0.25em;
}
sup {
  top: -0.5em;
}
img {
  border-style: none;
}
button,
input,
optgroup,
select,
textarea {
  font-family: inherit;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
}
button,
input {
  overflow: visible;
}
button,
select {
  text-transform: none;
}
[type="button"],
[type="reset"],
[type="submit"],
button {
  -webkit-appearance: button;
}
[type="button"]::-moz-focus-inner,
[type="reset"]::-moz-focus-inner,
[type="submit"]::-moz-focus-inner,
button::-moz-focus-inner {
  border-style: none;
  padding: 0;
}
[type="button"]:-moz-focusring,
[type="reset"]:-moz-focusring,
[type="submit"]:-moz-focusring,
button:-moz-focusring {
  outline: 1px dotted ButtonText;
}
fieldset {
  padding: 0.35em 0.75em 0.625em;
}
legend {
  box-sizing: border-box;
  color: inherit;
  display: table;
  max-width: 100%;
  padding: 0;
  white-space: normal;
}
progress {
  vertical-align: baseline;
}
textarea {
  overflow: auto;
}
[type="checkbox"],
[type="radio"] {
  box-sizing: border-box;
  padding: 0;
}
[type="number"]::-webkit-inner-spin-button,
[type="number"]::-webkit-outer-spin-button {
  height: auto;
}
[type="search"] {
  -webkit-appearance: textfield;
  outline-offset: -2px;
}
[type="search"]::-webkit-search-decoration {
  -webkit-appearance: none;
}
::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}
details {
  display: block;
}
summary {
  display: list-item;
}
template {
  display: none;
}
[hidden] {
  display: none;
}

*,
*::before,
*::after {
  box-sizing: inherit;
}

html {
  box-sizing: border-box;
  font-size: 20px;
  line-height: 1.5;
  -webkit-tap-highlight-color: transparent;
}

body {
  font-size: .8rem;
  overflow-x: hidden;
  text-rendering: optimizeLegibility;
}


h1 {
    font-size: 1.6rem;
  }

  h2 {
    font-size: 1.4rem;
  }

  h3 {
    font-size: 1.2rem;
  }

  h4 {
    font-size: 1rem;
  }
  h5 {
    font-size: 0.9rem;
  }

  h6 {
    font-size: 0.8rem;
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-weight: 700;
    line-height: 1.25;
    margin-bottom: 0.5em;
    margin-top: 0;
    font-family: ${(props: any) => props.theme.fonts.medium};
  }

  h1,
  h2,
  h3,
  h4 {
    color: ${colors.N50};
  }

  h5,
  h6 {
    color: ${colors.N60};
  }

  p {
    color: ${colors.N70};
    line-height: 1.5;
    margin-bottom: 0.5em;
    margin-top: 0;
    font-family: ${(props: any) => props.theme.fonts.normal};
  }

  a {
    font-family: ${(props: any) => props.theme.fonts.normal};
    color: ${colors.B50};
    outline: none;
    text-decoration: none;

    &:hover {
      color: ${colors.B30};
    }

    &:focus,
    &:hover,
    &:active {
      text-decoration: underline;
    }
    &:focus,
    &:active {
      color: ${colors.B60};
    }
    &:focus {
      text-decoration: none;
    }
  }

  b,
  strong {
    font-weight: bolder;
  }

  small {
    font-size: 80%;
    font-weight: 400;
    color: ${colors.N30};
  }

  blockquote {
    margin-left: 0;
    padding: 0.4rem 0.8rem;
    color: ${colors.N30};
    p:last-child {
      margin-bottom: 0;
    }
  }

  table {
    border-collapse: collapse;
    border-spacing: 0;
    text-align: left;
    width: 100%;
  }
  th {
    border-bottom: 0.05rem solid ${props => props.theme.colors["N5"]};
    padding: 0.6rem 0.4rem;
    border-bottom-width: 0.1rem;
  }
  td {
    border-bottom: 0.05rem solid ${props => props.theme.colors["N2"]};
    padding: 0.6rem 0.4rem;
  }
`;
