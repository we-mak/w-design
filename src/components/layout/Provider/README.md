Theme provider is based on [styled-components](https://www.styled-components.com/) and making theme with [styled-system](http://jxnblk.com/styled-system/)

Provider provides theme for your layout, you can supply your theme by insert your theme object by `theme` props
Default theme

```js static
defaultTheme = {
  breakpoint: ["24em", "32em", "48em", "64em", "80em"],
  fonts:
    '-apple-system, BlinkMacSystemFont, Helvetica, "Helvetica Neue", system-ui, sans-serif',
  fontSizes: [
    "0.7rem",
    "0.8rem",
    "0.9rem",
    "1rem",
    "1.2rem",
    "1.4rem",
    "1.6rem",
    "1.8rem",
    "2rem"
  ],
  fontWeights = { normal: 400, strong: 500, bold: 700, xbold: 900 },
  lineHeights = [1, 1.125, 1.25, 1.5],
  letterSpacings = { normal: "normal", caps: "0.25em" },
  radii = ["0rem", "0.1rem", "0.2rem", "0.3rem", "0.4rem"],
  space = [0, 4, 8, 16, 32, 64, 128], // space is used for margin and padding scales
  shadows = [
    "none",
    `0 .2rem .5rem ${hex2Rgba(colors.N50, 0.5)}`,
    `inset 0 0 0 1px rgba(87, 85, 217, .2), 0 0 4px rgba(87, 85, 217, .2)`
  ],
  colors: {...} // see colors
};
```

```js
<Provider>
  <h1>Default </h1>
  <p>
    Lorem ipsum
    <span>
      <a href="#">dolor </a>
    </span>
    sip amet
  </p>
</Provider>
```

```js
<Provider
  theme={{
    fonts: "Roboto, Helvetica, sans-serif"
  }}
>
  <h1>Roboto</h1>
  <p>
    Lorem ipsum
    <span>
      <a href="#">dolor </a>
    </span>
    sip amet
  </p>
</Provider>
```
