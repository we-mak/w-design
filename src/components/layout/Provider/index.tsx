import * as React from "react";
import { ThemeProvider } from "styled-components";
import defaultTheme from "../../../common/theme";
import { ResetCss } from "./ResetCss";
import Root from "./Root";

export interface Props extends React.HTMLProps<HTMLDivElement> {
  theme?: any;
}

class Provider extends React.Component<Props> {
  stylesheet: any;

  static displayName = "WProvider";

  public static defaultProps: Props = {
    theme: {}
  };

  componentDidMount() {
    this.stylesheet = document.createElement("style");
    this.stylesheet.type = "text/css";
    // this.stylesheet.setAttribute("rel", "preload");
    // this.stylesheet.setAttribute("as", "style");
    this.stylesheet.innerHTML = ResetCss;

    if (document && document.head) {
      document.head.appendChild(this.stylesheet);
    }
  }

  componentWillUnmount() {
    if (this.stylesheet && document && document.head) {
      document.head.removeChild(this.stylesheet);
      delete this.stylesheet;
    }
  }

  render() {
    const { theme, ...props } = this.props;
    return (
      <ThemeProvider theme={{ ...defaultTheme, ...theme }}>
        <Root {...props as any} />
      </ThemeProvider>
    );
  }
}

export default Provider;
