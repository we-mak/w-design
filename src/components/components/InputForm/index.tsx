/**
 * Input form
 * Input form with label, text hint...
 */
import * as React from "react";
import InputField from "../../elements/InputField";
import { InputFormProps } from "./types";
import { Label, FieldGroup, TextHint, RequiredMark } from "./Styled";

// TODO: add Icon

const InputForm: React.FunctionComponent<InputFormProps> = props => {
  const {
    id,
    label,
    hintMessage,
    isRequired,
    isError,
    isWarning,
    isSuccess,
    children,
    ...rest
  } = props;
  return (
    <FieldGroup>
      {label && (
        <Label htmlFor={id}>
          {label} {isRequired && <RequiredMark aria-hidden="true">*</RequiredMark>}
        </Label>
      )}
      <InputField
        fluid
        id={id}
        isError={isError}
        isWarning={isWarning}
        isValid={isSuccess}
        {...rest}
      />
      {children}
      {hintMessage && (
        <TextHint isError={isError} isWarning={isWarning} isSuccess={isSuccess}>
          {hintMessage}
        </TextHint>
      )}
    </FieldGroup>
  );
};

export default InputForm;
