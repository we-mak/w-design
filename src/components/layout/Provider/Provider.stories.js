import React from "react";
import { storiesOf } from "@storybook/react";
import Provider from "./Provider";

storiesOf("components/Provider", module).addWithJSX("default Provider", () => (
  <Provider>
    <h1>Default </h1>
    <p>
      Lorem ipsum
      <span>
        <a href="#"> dolor </a>
      </span>{" "}
      sip amet
    </p>
  </Provider>
));
