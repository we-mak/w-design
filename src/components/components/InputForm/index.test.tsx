import * as React from "react";
import { cleanup, render } from "react-testing-library";
import "jest-styled-components";
import InputForm from "./index";
import Provider from "../../layout/Provider";

afterEach(cleanup);

describe("<InputForm/>", () => {
  it("should render correctly", () => {
    const { container } = render(
      <Provider>
        <InputForm label="label" hintMessage="hint" />
        <InputForm isError />
        <InputForm isFocus />
        <InputForm isDisabled />
        <InputForm isRequired />
        <InputForm isWarning />
        <InputForm isSuccess />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  // Props test
  it("should have required attribute", () => {
    // const { container } = render(
    //   <Provider>
    //     <InputForm isRequired />
    //   </Provider>
    // );
  });

  it("should have label props", () => {
    const { getByText } = render(
      <Provider>
        <InputForm label="Hello" />
      </Provider>
    );
    getByText("Hello");
  });

  it("should have hint message", () => {
    const { getByText } = render(
      <Provider>
        <InputForm hintMessage="Hello" />
      </Provider>
    );
    getByText("Hello");
  });
});
