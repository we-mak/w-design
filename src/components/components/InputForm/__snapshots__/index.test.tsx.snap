// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputForm/> should render correctly 1`] = `
.c3 {
  background: transparent;
  background-image: none;
  outline: none;
  max-width: 100%;
  display: block;
  position: relative;
  border: 0.05rem solid #BECEDC;
  border-radius: 0.2rem;
  color: #202B33;
  padding: 0.35rem 0.4rem;
  font-size: 0.8rem;
  font-family: inherit;
  height: 1.8rem;
  width: 100%;
  line-height: 1.125;
  -webkit-transition: background 0.2s,border 0.2s,box-shadow 0.2s,color 0.2s;
  transition: background 0.2s,border 0.2s,box-shadow 0.2s,color 0.2s;
}

.c3:focus {
  box-shadow: 0 0 0 0.1rem rgba(0,140,192,0.2);
  border-color: #008CC0;
}

.c3::-webkit-input-placeholder {
  color: #90A9BF;
}

.c3::-moz-placeholder {
  color: #90A9BF;
}

.c3:-ms-input-placeholder {
  color: #90A9BF;
}

.c3::placeholder {
  color: #90A9BF;
}

.c3:disabled,
.c3[disabled] {
  cursor: not-allowed;
  color: #90A9BF !important;
  background: #EFF3F6 !important;
}

.c3[type="file"] {
  height: auto;
}

.c5 {
  background: transparent;
  background-image: none;
  outline: none;
  max-width: 100%;
  display: block;
  position: relative;
  border: 0.05rem solid #DC1F1F;
  border-radius: 0.2rem;
  color: #202B33;
  padding: 0.35rem 0.4rem;
  font-size: 0.8rem;
  font-family: inherit;
  height: 1.8rem;
  width: 100%;
  line-height: 1.125;
  -webkit-transition: background 0.2s,border 0.2s,box-shadow 0.2s,color 0.2s;
  transition: background 0.2s,border 0.2s,box-shadow 0.2s,color 0.2s;
}

.c5:focus {
  box-shadow: 0 0 0 0.1rem rgba(220,31,31,0.2);
  border-color: #DC1F1F;
}

.c5::-webkit-input-placeholder {
  color: #90A9BF;
}

.c5::-moz-placeholder {
  color: #90A9BF;
}

.c5:-ms-input-placeholder {
  color: #90A9BF;
}

.c5::placeholder {
  color: #90A9BF;
}

.c5:disabled,
.c5[disabled] {
  cursor: not-allowed;
  color: #90A9BF !important;
  background: #EFF3F6 !important;
}

.c5[type="file"] {
  height: auto;
}

.c8 {
  background: transparent;
  background-image: none;
  outline: none;
  max-width: 100%;
  display: block;
  position: relative;
  border: 0.05rem solid #FA9B00;
  border-radius: 0.2rem;
  color: #202B33;
  padding: 0.35rem 0.4rem;
  font-size: 0.8rem;
  font-family: inherit;
  height: 1.8rem;
  width: 100%;
  line-height: 1.125;
  -webkit-transition: background 0.2s,border 0.2s,box-shadow 0.2s,color 0.2s;
  transition: background 0.2s,border 0.2s,box-shadow 0.2s,color 0.2s;
}

.c8:focus {
  box-shadow: 0 0 0 0.1rem rgba(250,155,0,0.2);
  border-color: #FA9B00;
}

.c8::-webkit-input-placeholder {
  color: #90A9BF;
}

.c8::-moz-placeholder {
  color: #90A9BF;
}

.c8:-ms-input-placeholder {
  color: #90A9BF;
}

.c8::placeholder {
  color: #90A9BF;
}

.c8:disabled,
.c8[disabled] {
  cursor: not-allowed;
  color: #90A9BF !important;
  background: #EFF3F6 !important;
}

.c8[type="file"] {
  height: auto;
}

.c1 {
  position: relative;
  font-family: Helvetica,Arial,sans-serifpx;
}

.c1:not(:last-child) {
  margin-bottom: 8px;
}

.c1 i {
  color: #6C869B;
}

.c1 i.icon-before {
  height: 0.8rem;
  margin: 0 0.25rem;
  position: absolute;
  top: 73%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 0.8rem;
  z-index: 2;
  left: 0.1rem;
}

.c1 i.icon-after {
  height: 0.8rem;
  margin: 0 0.25rem;
  position: absolute;
  top: 73%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 0.8rem;
  z-index: 2;
  right: 0.1rem;
}

.c1 input.icon-before {
  padding-left: 1.3rem !important;
}

.c1 input.icon-after {
  padding-right: 1.3rem !important;
}

.c2 {
  display: block;
  line-height: 1;
  padding: 8px 0;
  font-weight: 500;
}

.c4 {
  display: block;
  margin-top: 0.4rem;
  color: #566E82;
}

.c6 {
  display: block;
  margin-top: 0.4rem;
  color: #DC1F1F;
}

.c9 {
  display: block;
  margin-top: 0.4rem;
  color: #FA9B00;
}

.c10 {
  display: block;
  margin-top: 0.4rem;
  color: #29A634;
}

.c7 {
  color: #DC1F1F;
  font-size: 0.7rem;
  top: -0.2rem;
}

.c0 h1 {
  font-size: 1.6rem;
}

.c0 h2 {
  font-size: 1.4rem;
}

.c0 h3 {
  font-size: 1.2rem;
}

.c0 h4 {
  font-size: 1rem;
}

.c0 h5 {
  font-size: 0.9rem;
}

.c0 h6 {
  font-size: 0.8rem;
}

.c0 h1,
.c0 h2,
.c0 h3,
.c0 h4,
.c0 h5,
.c0 h6 {
  font-weight: 700;
  line-height: 1.25;
  margin-bottom: 0.5em;
  margin-top: 0;
  font-family: Helvetica,Arial,sans-serif;
}

.c0 h1,
.c0 h2,
.c0 h3,
.c0 h4 {
  color: #394B59;
}

.c0 h5,
.c0 h6 {
  color: #293742;
}

.c0 p {
  color: #202B33;
  line-height: 1.5;
  margin-bottom: 0.5em;
  margin-top: 0;
  font-family: Helvetica,Arial,sans-serif;
}

.c0 a {
  font-family: Helvetica,Arial,sans-serif;
  color: #008CC0;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a:hover {
  color: #00A7E4;
}

.c0 a:focus,
.c0 a:hover,
.c0 a:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 a:focus,
.c0 a:active {
  color: #007099;
}

.c0 a:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 b,
.c0 strong {
  font-weight: bolder;
}

.c0 small {
  font-size: 80%;
  font-weight: 400;
  color: #566E82;
}

.c0 blockquote {
  margin-left: 0;
  padding: 0.4rem 0.8rem;
  color: #566E82;
}

.c0 blockquote p:last-child {
  margin-bottom: 0;
}

.c0 table {
  border-collapse: collapse;
  border-spacing: 0;
  text-align: left;
  width: 100%;
}

.c0 th {
  border-bottom: 0.05rem solid #BECEDC;
  padding: 0.6rem 0.4rem;
  border-bottom-width: 0.1rem;
}

.c0 td {
  border-bottom: 0.05rem solid #EFF3F6;
  padding: 0.6rem 0.4rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2"
      >
        label
         
      </label>
      <input
        autocomplete="off"
        class="c3"
        type="text"
        value=""
      />
      <small
        class="c4"
      >
        hint
      </small>
    </div>
    <div
      class="c1"
    >
      <label
        class="c2"
      >
         
      </label>
      <input
        autocomplete="off"
        class="c5"
        type="text"
        value=""
      />
      <small
        class="c6"
      />
    </div>
    <div
      class="c1"
    >
      <label
        class="c2"
      >
         
      </label>
      <input
        autocomplete="off"
        class="c3"
        type="text"
        value=""
      />
      <small
        class="c4"
      />
    </div>
    <div
      class="c1"
    >
      <label
        class="c2"
      >
         
      </label>
      <input
        autocomplete="off"
        class="c3"
        disabled=""
        type="text"
        value=""
      />
      <small
        class="c4"
      />
    </div>
    <div
      class="c1"
    >
      <label
        class="c2"
      >
         
        <sup
          aria-hidden="true"
          class="c7"
        >
          *
        </sup>
      </label>
      <input
        autocomplete="off"
        class="c3"
        type="text"
        value=""
      />
      <small
        class="c4"
      />
    </div>
    <div
      class="c1"
    >
      <label
        class="c2"
      >
         
      </label>
      <input
        autocomplete="off"
        class="c8"
        type="text"
        value=""
      />
      <small
        class="c9"
      />
    </div>
    <div
      class="c1"
    >
      <label
        class="c2"
      >
         
      </label>
      <input
        autocomplete="off"
        class="c3"
        type="text"
        value=""
      />
      <small
        class="c10"
      />
    </div>
  </div>
</div>
`;
